runtime/src/main/java/com/flipkart/flux/representation/ClientElbPersistenceService.java

```diff
@@ -0,0 +1,277 @@
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.transaction.Transactional;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * <code>ClientElbPersistenceService</code> class does persistence related operations in memory
+ * before storing, updating, reading or deleting objects in DB. It is also responsible for maintaining
+ * an in-memory cache of clientElb entries for faster query results.
+ * @author akif.khan
+ */
+@Singleton
+public class ClientElbPersistenceService {
+
+    private ClientElbDAO clientElbDAO;
+
+    private ConcurrentHashMap<String, ClientElb> clientElbCache;
+
+    private ConcurrentHashMap<String, Long> lastAccessed;
+
+    private Integer MAX_CACHE_SIZE;
+
+    private static final Logger logger = LoggerFactory.getLogger(ClientElbPersistenceService.class);
+
+    @Inject
+    public ClientElbPersistenceService(ClientElbDAO clientElbDAO,
+                                       @Named("elbCache.maxSize") Integer MAX_CACHE_SIZE) {
+
+        this.clientElbDAO = clientElbDAO;
+        this.MAX_CACHE_SIZE = MAX_CACHE_SIZE;
+        this.clientElbCache = new ConcurrentHashMap<>();
+        this.lastAccessed = new ConcurrentHashMap<>();
+    }
+
+    public Integer getClientElbCacheSize() {
+        return clientElbCache.size();
+    }
+
+    public boolean clientElbCacheContainsKey(String key) {
+        return clientElbCache.containsKey(key);
+    }
+
+    public String getClientElbCacheUrl(String key) {
+        return clientElbCache.get(key).getElbUrl();
+    }
+
+    /**
+     * Query Local Cache ClientElbCache for a given clientId
+     * Updates and stores last access time of a Client ELb Entry in lastAccessed Map which is a basis for
+     * LRU operation in case of cache entry replacement.
+     * @param clientId
+     * @return clientElb
+     */
+    private ClientElb queryClientElbCache(String clientId) {
+
+        ClientElb clientElb = null;
+        try {
+            clientElb = clientElbCache.get(clientId);
+            if (clientElb != null) {
+                lastAccessed.replace(clientId, System.currentTimeMillis());
+            }
+        }
+        catch(Exception ex) {
+            logger.error("Exception in ClientElbCache read {} {}", ex.getMessage(), ex.getStackTrace());
+        }
+        return clientElb;
+
+    }
+
+    /**
+     * Refreshes ClientElbCache in case of cache miss, either with a new entry addition into the cache if cache has not
+     * reached it's maximum size or a LRU replacement policy.
+     * It's a thread safe method synchronized on singleton instance of this class.
+     * @param clientId
+     * @param clientElb
+     */
+    private synchronized void clientElbCacheRefresh(String clientId, ClientElb clientElb) {
+
+        try {
+
+            if(clientElbCache.size() < MAX_CACHE_SIZE) {
+                clientElbCache.put(clientId, clientElb);
+                lastAccessed.put(clientId, System.currentTimeMillis());
```

Comment:
why not use GuavaCache or something like that instead of reinventing the wheel?

Reply:
Introduced Google Guava Cache.
See Commit title: Replace customized LRU cache with Google Guava Cache...